AcademicFlow.Domain.Contracts/IServices/IProgramService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using AcademicFlow.Domain.Contracts.Entities;
+
+namespace AcademicFlow.Domain.Contracts.IServices
+{
+    public interface IProgramService
+    {
+        int? AddProgram(Program entity);
+        Program? GetProgramById(int id);
+        void UpdateProgram(Program entity);
+        IQueryable<Program> GetAll();
+        void DeleteProgram(int id);
```

Comment:
why the method is not async? look at other services, it should be async and return Task